core/chain-configs/src/genesis_config.rs

```diff
@@ -3,23 +3,22 @@
 //! NOTE: chain-configs is not the best place for `GenesisConfig` since it
 //! contains `RuntimeConfig`, but we keep it here for now until we figure
 //! out the better place.
-use std::fs::File;
-use std::io::{Read, Write};
-use std::path::{Path, PathBuf};
+use std::marker::PhantomData;
```

Comment:
/cc @nearmax @evgenykuzyakov @bowenwang1996 @mikhailOK I had completely rewritten the `impls` for `GenesisConfig` (and introduced `Genesis` and `GenesisRecords` structs). It seems to be quite clean and minimal, but I invite you to eyeball the changes if there is anything odd. I find the new design much better as splitting the records from the rest of the structure was a mess (`genesis_config` RPC implementation has been simplified to a single line after this refactoring).